server/src/main/java/org/opensearch/indices/recovery/RemoteRecoveryTargetHandler.java

```diff
@@ -106,7 +85,11 @@ public RemoteRecoveryTargetHandler(
         Consumer<Long> onSourceThrottle
     ) {
         this.transportService = transportService;
-        this.threadPool = transportService.getThreadPool();
+        this.retryableTransportClient = new RetryableTransportClient(
```

Comment:
Nitpick - add a comment to clarify why it's safe to cache the timeout value as opposed to the previous behavior of creating a new Options object for each method call